examples/DivisorCurve.jl

```diff
@@ -0,0 +1,329 @@
+"""
+struct ProjCurveDivisor{S <: FieldElem} <: CurveDivisor
+    C::ProjPlaneCurve{S}
+    divisor::Dict{Oscar.Geometry.ProjSpcElem{S}, Int}
+    degree::Int
+    function ProjCurveDivisor(C::ProjPlaneCurve{S}, D::Dict{Oscar.Geometry.ProjSpcElem{S}, Int}) where S <: FieldElem
+        for P in keys(D)
+            ison_curve(P, C) || error("The point ", P, " is not on the curve")
+            if iszero(D[P])
+                delete!(D, P)
+            end
+        end
+        new{S}(C,
+            D,
+            sum(values(D)))
+    end
+end
+#function Base.show(io::IO, D::ProjCurveDivisor)
+#    println(io, "On the curve ", D.C.eq, ", divisor defined by ", D.divisor)
+#end
+
+function ProjCurveDivisor(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}, m::Int=1) where S <: FieldElem
+    return ProjCurveDivisor(C, Dict(P => m))
+end
+
+################################################################################
+# To get a nice output for the divisors. The needs_parentheses is needed since
+# expressify wants to know if parentheses are needed.
+
+function AbstractAlgebra.needs_parentheses(A::Array{T, 1}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::AffineCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (p, i) in a.divisor
+      ep = AbstractAlgebra.expressify(p.coord, context = context)
+      push!(prod.args, Expr(:call, :*, i, ep))
+   end
+   return prod
+end
+
+function AbstractAlgebra.needs_parentheses(A::Oscar.Geometry.ProjSpcElem{T}) where T <: RingElem
+    return false
+end
+
+function AbstractAlgebra.expressify(@nospecialize(a::ProjCurveDivisor); context = nothing)
+   prod = Expr(:call, :+)
+   for (p, i) in a.divisor
+      ep = AbstractAlgebra.expressify(p, context = context)
+      push!(prod.args, Expr(:call, :*, i, ep))
+   end
+   return prod
+end
+
+function Base.show(io::IO, ::MIME"text/plain", a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+function Base.show(io::IO, a::CurveDivisor)
+  print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+
+################################################################################
+################################################################################
+
+function curve(D::CurveDivisor)
+    return D.C
+end
+
+function Oscar.degree(D::CurveDivisor)
+    return D.degree
+end
+
+################################################################################
+
+function _check_same_curve(D::CurveDivisor, E::CurveDivisor)
+    return D.C == E.C
+end
+
+################################################################################
+################################################################################
+# Sum of two divisors
+
+function Base.:+(D::AffineCurveDivisor, E::AffineCurveDivisor)
+    _check_same_curve(D, E) || error("The divisors are not on the same curve")
```

Comment:
Optional comment:

I actually meant to have the error in `_check_same_curve`, to ensure it looks the same everywhere. So I'd either do that, or revert it back to 
```suggestion
    curve(D) == curve(E) || error("The divisors are not on the same curve")
```
Or leave it as it is now, it's not super important, I don't mean to drag you down with this :-)

Reply:
OK, I just changed it !